archive/imo1988_q6.lean

```diff
@@ -0,0 +1,282 @@
+/-
+Copyright (c) 2019 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import data.rat.basic
+import tactic
+
+/-!
+# IMO1988 Q6 and constant descent Vieta jumping
+
+Question 6 of IMO1988 is somewhat (in)famous. Several expert problem solvers
+could not tackle the question within the given time limit.
+The problem lead to the introduction of a new proof technique,
+so called “Vieta jumping”.
+
+In this file we formalise constant descent Vieta jumping,
+and apply this to prove Q6 of IMO1988.
+To illustrate the technique, we also prove a similar result.
+
+-/
+
+-- open_locale classical
+
+local attribute [instance] classical.prop_decidable
+local attribute [simp] nat.pow_two
+
+/-- Constant descent Vieta jumping is a proof technique
+that can typically be used to show that certain algebraic curves of degree 2
+in the affine plane do not have integral points (or only an explicit set of solutions).
+
+We need a long list of assumptions for this lemma.
+The upside is that it takes care of several annoying edge cases.
+First of all, the user should provide x and y
+that satisfy the predicate H(x,y).
```

Comment:
I had a hard time following this, and I already know roughly how to solve the IMO problem. I think it would help to give as an example the particular H(x,y) which is used for the IMO problem. Also, you never mention `claim`, which is pretty important (at least if you are trying to understand the lemma just from the docstring, and not from looking at its type).

Reply:
I have thoroughly rewritten the docstring. I hope it is better now.